File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260917-060430-47161/output/project/src/main/java/surveycrawler/FirmSand.java
package surveycrawler;

public class FirmSand extends GroundType {

    // moving

    @Override
    public void moveTowards(SurveyCrawler aCrawler, Point aDirection) {
        aCrawler.moveOnFirmSandTowards(aDirection);
    }

    // facing

    @Override
    public void assertCanTurn(SurveyCrawler aCrawler) {
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-210220-97414/analysis/own-tests-project/src/surveycrawler/firm_sand.py
from surveycrawler.ground import Ground


class FirmSand(Ground):

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        a_crawler.move_cells_towards(1, a_direction)

    # testing

    def assert_can_be_occupied(self):
        pass  # firm sand can always be occupied


